src/clojure_tic_tac_toe/minimax.clj

```diff
@@ -0,0 +1,59 @@
+(ns clojure-tic-tac-toe.minimax
+  (:require [clojure-tic-tac-toe.game_handler :as game_handler]
+            [clojure-tic-tac-toe.win_checker :as win_checker]))
+
+(defn- generate-boards-with-moves
+  [board sorted-moves next-player]
+  (map #(game_handler/add-move board % next-player) sorted-moves))
+
+(defn- get-scores
+  [scores-and-moves]
+  (map #(:score %) scores-and-moves))
+
+
+(defn- max-or-min-based-on-player
+  [player]
+  (cond
+    (= player :X) max
+    (= player :O) min))
+
+(defn- get-score-based-on-player
+  [scores player]
+  (apply (max-or-min-based-on-player player) scores))
+
+
+(defn- create-map-of-moves-and-scores
+  [moves scores]
+  (zipmap moves scores))
+
+(defn- get-pairs-with-target-score
+  [moves-and-scores target-score]
+  (filter #(-> % val (= target-score)) moves-and-scores))
+
+(defn- get-move-from-pairs
+  [pairs]
+  (first (first pairs)))
+
+(defn- get-move-based-on-score
+  [moves scores target-score]
+  ((comp get-move-from-pairs
+         #(get-pairs-with-target-score % target-score)
+         create-map-of-moves-and-scores)
+   moves scores))
+
+(defn minimax-move-and-score
+  [board next-player]
+  (if (game_handler/is-game-finished? board)
+    (let [score (win_checker/calculate-score board)]
+      {:move :invalid-move, :score score})
+    (let [sorted-moves (game_handler/get-sorted-moves board)
```

Comment:
What is the relevance of the sorting?  Why does your algorithm care?

Reply:
I mainly sorted the moves to make it easier to follow what the program is doing, and make it easier to debug. However, having the moves be in random order will improve the behavior of the computer (the computer won't always move to the same spot, given the same board).